ntbs-service-unit-tests/DataAccess/DataQualityRepositoryTests.cs

```diff
@@ -0,0 +1,94 @@
+
+        public DataQualityRepositoryTests()
+        {
+            ContextOptions = new DbContextOptionsBuilder<NtbsContext>()
+                .UseInMemoryDatabase(nameof(DataQualityRepositoryTests))
+                .Options;
+        }
+
+        private DbContextOptions<NtbsContext> ContextOptions { get; }
+
+        [Fact]
+        public async Task FiltersOutNotificationsWithTheAlertTypeAlreadyPresent()
+        {
+            // ARRANGE
+            using (var context = new NtbsContext(ContextOptions))
+            {
+                await context.Database.EnsureDeletedAsync();
+                await context.Database.EnsureCreatedAsync();
+                
+                // We will use the birth country alert as a proxy for testing the common path ways of the
+                // data quality repository
+                // All seeded notifications will be eligible for this alert
+                
+                var notification1 = NotificationEligibleForCountryOfBirthDqAlert(NoAlertsName);
+
+                var notification2 = NotificationEligibleForCountryOfBirthDqAlert(BirthCountryAlertName);
+                notification2.Alerts = new List<Alert> {new DataQualityBirthCountryAlert()};
+
+                var notification3 = NotificationEligibleForCountryOfBirthDqAlert(ClinicalDatesAlertName);
+                notification3.Alerts = new List<Alert> {new DataQualityClinicalDatesAlert()};
+
+                await context.AddRangeAsync(notification1, notification2, notification3);
+                
+                await context.SaveChangesAsync();
+            }
+            
+            using (var context = new NtbsContext(ContextOptions))
+            {
+                var repo = new DataQualityRepository(context);
+                
+                // ACT
+                // Get all possible ones
+                var notifications = await repo.GetNotificationsEligibleForDqBirthCountryAlertsAsync(100, 0);
+                
+                // ASSERT
+                // Out of the 3 eligible notifications, 1 already has the alert
+                Assert.True(notifications.Any(n => n.PatientDetails.FamilyName == NoAlertsName), 
+                    "Eligible notification with no other alerts not selected");
+                Assert.True(notifications.Any(n => n.PatientDetails.FamilyName == ClinicalDatesAlertName),
+                    "Eligible notification with only other alerts types not selected");
+                Assert.Equal(2, notifications.Count);
+            }
+        }
+        
+        private static Notification NotificationEligibleForCountryOfBirthDqAlert(string name)
+        {
+            return new Notification
+            {
+                NotificationStatus = NotificationStatus.Notified,
+                NotificationDate = DateTime.Now
+                    .AddDays(-DataQualityRepository.MinNumberDaysNotifiedForAlert)
+                    .AddDays(-1),
```

Comment:
it took me a few seconds to realise what this -1 day is for :)

Reply:
hmm, good point, I took a few seconds too even though it's been a day since i wrote this :P